### PR TITLE
Definitions for cbor-js

### PR DESCRIPTION
**Pull request by Longqin88888**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/cbor-js/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "private": true,
+  "name": "@types/cbor-js",
+  "version": "0.1.0",
```
Comment (codershiba) [inline types/cbor-js/package.json:4]:
```suggestion
  "version": "0.1.9999",
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/cbor-js/index.d.ts
1. types/cbor-js/package.json
1. types/cbor-js/cbor-js-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a8ee56df26a8ab6eb4c49937ff5b79fb5b0b8754
</p>

File: types/cbor-js/cbor-js-tests.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import CBOR from "cbor-js"
+
+var initial = { Hello: "World" };
+var encoded = CBOR.encode(initial);
+var decoded = CBOR.decode(encoded);
```
Comment (peterblazejewicz) [inline types/cbor-js/cbor-js-tests.ts:5]:
```suggestion
let initial = { Hello: "World" };
let encoded = CBOR.encode(initial);
let decoded = CBOR.decode(encoded);
```
(it's 2025)

File: types/cbor-js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Type definitions for cbor-js 0.1.0
+// Project: https://github.com/paroga/cbor-js
+
+export function encode(data: any): ArrayBuffer;
+
+export function decode(encoded: ArrayBuffer): any;
```
Comment (peterblazejewicz) [inline types/cbor-js/index.d.ts:6]:
```suggestion
export function decode(encoded: ArrayBuffer): any;

export as namespace CBOR;
```
https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library
https://github.com/paroga/cbor-js/blob/master/cbor.js#L399 (and others)

File: types/cbor-js/cbor-js-tests.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import CBOR from "cbor-js"
```
Comment (peterblazejewicz) [inline types/cbor-js/cbor-js-tests.ts:1]: you would eventually need this:
```suggestion
import * as CBOR from "cbor-js"
```

File: types/cbor-js/package.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    "projects": [
+        "https://github.com/paroga/cbor-js"
+    ],
+    "devDependencies": {
+        "@types/cbor-js": "workspace:."
+    },
+    "owners": [
+        {
+            "name": "long2014",
+            "githubUsername": "long2014"
```
Comment (codershiba) [inline types/cbor-js/package.json:14]:
```suggestion
            "githubUsername": "Longqin88888"
```

Comment (codershiba) [reply types/cbor-js/package.json:14]: Username is incorrect

**Review by sheetalkamat — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by Longqin88888** [comment]
Ready to merge

**Pull request merged**